docs: What’s New & Upgrade Guide 10.1 AuthNZ (#72000)
grafana/grafana: docs/sources/whatsnew/whats-new-in-v10-1.md

The `accessTokenExpirationCheck` feature toggle will be removed in Grafana v10.2.
{{% /admonition %}}

### OAuth role mapping enforcement

> <!-- AuthNZ -->

Reviewer: We can fall back to the team, but we always want a specific person to be the point of contact if it's needed during the ongoing editing process. 

```suggestion
<!-- Jo Guerreiro, AuthNZ -->
```

> 

Reviewer: Which editions of Grafana does this apply to: Enterprise, OSS, Cloud?

Reviewer 2: All 3! I added more context at line 68.

Reviewer: Okay I've moved and rewritten the note to be consistent with the rest of the What's new. We need the same information for the other two features as well.

Reviewer 2: Changed!

> This change impacts `GitHub` OAuth, `Gitlab` OAuth, `Okta` OAuth and `Generic` OAuth.

Reviewer: Are `GitHub`, `GitLab`, etc configuration options or UI selections here? I just want to determine if this formatting is appropriate.

Reviewer 2: These are configuration options.

Reviewer: Okay, formatting is fine then.

Currently if no organization role mapping is found for a user when connecting via OAuth, Grafana doesn’t update the user’s organization role.

> With Grafana 10.1, on every login, if the role_attribute_path property does not return a role, then the user is assigned the role specified by the auto_assign_org_role option or the default role for the organization, by default, Viewer.

Reviewer: Can the default role for an org be changed? If so then we might want to reword this a bit to clarify.
```suggestion
With Grafana 10.1, on every login, if the `role_attribute_path` property does not return a role, then the user is assigned the role specified by the `auto_assign_org_role` option or the default role for the organization, by default, Viewer.
```

> To avoid overriding manually set roles, enable the skip_org_role_sync option in the Grafana configuration for your OAuth provider before the user logs in for the first time.

Reviewer: Do we mean a specific user or any user?
```suggestion
To avoid overriding manually set roles, enable the `skip_org_role_sync` option in the Grafana configuration for your OAuth provider before the user logs in for the first time.
```

### GitLab OIDC support

> <!-- AuthNZ -->

Reviewer: ```suggestion
<!-- Jo Guerreiro, AuthNZ -->
```

> Grafana 10.1 now supports GitLab OIDC via the `GitLab` OAuth provider in addition to the existing `GitLab` OAuth2 provider. This allows you to use GitLab OIDC to authenticate users in Grafana.

Reviewer: ```suggestion
Grafana 10.1 now supports GitLab OIDC through the `GitLab` OAuth provider in addition to the existing `GitLab` OAuth2 provider. This allows you to use GitLab OIDC to authenticate users in Grafana.
```

This allows Grafana to reduce the access scope to only the required scopes for authentication and authorization instead
of full read API access.

To learn how to migrate your GitLab OAuth2 setup to OIDC, refer to our [GitLab authentication documentation]({{< relref "../setup-grafana/configure-security/configure-authentication/gitlab/" >}}).

> ### Google OIDC and teamsync support

Reviewer 3: ```suggestion
### Google OIDC and Team Sync support
```

> <!-- AuthNZ -->

Reviewer: ```suggestion
<!-- Jo Guerreiro, AuthNZ -->
```

> Grafana 10.1 now supports Google OIDC via the `Google` OAuth provider in addition to the existing `Google` OAuth2 provider. This allows you to use Google OIDC to authenticate users in Grafana.

Reviewer: Same question as line 67? What is the formatting trying to communicate?

> Grafana 10.1 now supports Google OIDC via the `Google` OAuth provider in addition to the existing `Google` OAuth2 provider. This allows you to use Google OIDC to authenticate users in Grafana.

Reviewer: ```suggestion
Grafana 10.1 now supports Google OIDC through the `Google` OAuth provider in addition to the existing `Google` OAuth2 provider. This allows you to use Google OIDC to authenticate users in Grafana, which in turn lets Grafana reduce the access scope to only the required scopes for authentication and authorization.
.
```

> This allows Grafana to reduce the access scope to only the required scopes for authentication and authorization.
> 

Reviewer: ```suggestion
```

> This release also adds support for Google OIDC in Team Sync. You can now easily add users to teams by utilizing their Google groups.

Reviewer: ```suggestion
This release also adds support for Google OIDC in Team Sync. You can now easily add users to teams by using their Google groups.
```

> To learn how to migrate your Google OAuth2 setup to OIDC and how to setup teamsync, refer to our [Google authentication documentation]({{< relref "../setup-grafana/configure-security/configure-authentication/google/" >}}).

Reviewer 3: Nit: I'm not actually sure what our preferred way of referring to team sync is, but it would be good to at least be consistent within the section. I'd go with `Team Sync` (and also update the section header).

```suggestion
To learn how to migrate your Google OAuth2 setup to OIDC and how to setup Team Sync, refer to our [Google authentication documentation]({{< relref "../setup-grafana/configure-security/configure-authentication/google/" >}}).
```

> To learn how to migrate your Google OAuth2 setup to OIDC and how to setup Team Sync, refer to our [Google authentication documentation]({{< relref "../setup-grafana/configure-security/configure-authentication/google/" >}}).

Reviewer: ```suggestion
To learn how to migrate your Google OAuth2 setup to OIDC and how to set up Team Sync, refer to our [Google authentication documentation]({{< relref "../setup-grafana/configure-security/configure-authentication/google/" >}}).
```

#### Adds support for toggling common labels

Adds support for toggling common labels that are shared between all alert instances. You can toggle between showing or hiding them for each individual alert instance.

> 

Reviewer: ```suggestion

## Authentication and authorization

```

### OAuth role mapping enforcement

<!-- Jo Guerreiro, AuthNZ -->

> 

Reviewer: ```suggestion

_Generally available in Grafana Open Source, Enterprise, and Cloud._

```

This change impacts `GitHub` OAuth, `Gitlab` OAuth, `Okta` OAuth and `Generic` OAuth.

> This change impacts the Enterprise, OSS and cloud versions of Grafana.

Reviewer: ```suggestion
```
